File: Algorithms/1010.Pairs-of-Songs-With-Total-Durations-Divisible-by-60/pairs_of_songs_with_total_durations_divisible_by_60.py
import collections
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        res = 0
        cntMap = collections.Counter()
        for ele in time:
            res += cntMap[60 - ele % 60] if ele % 60 else cntMap[0]
            cntMap[ele % 60] += 1
        return res

File: Algorithms/1010.Pairs-of-Songs-With-Total-Durations-Divisible-by-60/pairs_of_songs_with_total_durations_divisible_by_60.go
func numPairsDivisibleBy60(time []int) int {
    res := 0
    cntMap := make([]int, 60)
    for _, ele := range time {
        res += cntMap[(60 - ele % 60) % 60]
        cntMap[ele % 60] += 1
    }
    return res
}